### ISSUE TITLE
no such instruction: `v4fmaddps (%rax),%zmm0,%zmm4'

### ISSUE THREAD

**Issue by J-cztery** [mattgodbolt/compiler-explorer#554]
See https://godbolt.org/g/R5pm1P: 

~~~
#include` <immintrin.h>
__m512 knm(__m512 Vdata[4]){
        __m128 c;
        return  _mm512_4fmadd_ps(
                                _mm512_setzero_ps( ),
                                Vdata[0],
                                Vdata[1],
                                Vdata[2],
                                Vdata[3],
                                &c);  
}
~~~

With compiler flags `-mavx5124fmaps` and compiler gcc 7.2 it outputs: 
~~~
/tmp/ccMGJEbN.s: Assembler messages:
/tmp/ccMGJEbN.s:64: Error: no such instruction: `v4fmaddps (%rax),%zmm0,%zmm4'
Compiler exited with result code 1
~~~

**Comment by mattgodbolt** [comment]
A very strange one, the compiler emitting an instruction the assembler can't handle. I build the assembler alongside the compiler, so they should be in step.

**Comment by mattgodbolt** [comment]
Looks like I need to upgrade binutils. Will try with the nightly tonight and check tomorrow. If the nightly GCC starts working I'll rebuild the 7.x compilers.

**Comment by Mandar12** [comment]
Both gcc 7.1 and 7.2 has this error (-mavx5124fmaps is not available for gcc 6 and older).
gcc trunk is compiling successfully [[https://godbolt.org/g/3uMvUQ]](https://godbolt.org/g/3uMvUQ) But without v4fmaddps instruction.

Strangely on my windows machine, gcc 7.2 compiles same code successfully with v4fmaddps instruction.

**Comment by mattgodbolt** [comment]
@Mandar12 I rebuilt the latest GCC with the latest binutils. It now compiles the instruction: but the `objdump` command doesn't recognise it, hence in your link there's an opcode that says ` (bad)  `.

I'll update the other GCCs at some point soon. I'm surprised at `objdump` - I build that alongside `gas` etc so it seems like either I'm running the wrong `objdump` or it doesn't support it.

**Comment by Mandar12** [comment]
@mattgodbolt looks like binutils issue. On my ubuntu setup, binutils version is 2.26.1. While on my windows setup, binutils version is 2.29.1.

**Comment by mattgodbolt** [comment]
(I have'nt made progress on this, so unassigning)

**Comment by mattgodbolt** [comment]
Fixed since GCC 7.3 -- let me know if you need it in older versions of GCC.
